execution_engine/src/system/mint.rs

```diff
@@ -63,29 +69,7 @@ pub trait Mint: RuntimeProvider + StorageProvider + SystemProvider {
             return Err(Error::InvalidTotalSupplyReductionAttempt);
         }
 
-        if amount.is_zero() {
-            return Ok(()); // no change to supply
-        }
-
-        // get total supply or error
-        let total_supply_uref = match self.get_key(TOTAL_SUPPLY_KEY) {
-            Some(Key::URef(uref)) => uref,
-            Some(_) => return Err(Error::MissingKey), // TODO
-            None => return Err(Error::MissingKey),
-        };
-        let total_supply: U512 = self
-            .read(total_supply_uref)?
-            .ok_or(Error::TotalSupplyNotFound)?;
-
-        // decrease total supply
-        let reduced_total_supply = total_supply
-            .checked_sub(amount)
-            .ok_or(Error::ArithmeticOverflow)?;
-
-        // update total supply
-        self.write(total_supply_uref, reduced_total_supply)?;
-
-        Ok(())
+        reduce_total_supply_unchecked(self, amount)
```

Comment:
As a rule of thumb, we usually don't import functions but qualify them with the module
```suggestion
        detail::reduce_total_supply_unchecked(self, amount)
```